theme/static/js/hs_file_browser.js

```diff
@@ -1080,19 +1080,14 @@ $(document).ready(function () {
 
     function onPaste() {
         var folderName = $("#fb-files-container li.ui-selected").children(".fb-file-name").text();
-        var targetPath = $("#hs-file-browser").attr("data-current-path");
-
-        if (folderName && folderName.lastIndexOf(".") == -1) {  // Makes sure the destination is a folder
```

Comment:
Also note that this onPaste method handles pasting from the UI buttons and not the Drag and Drop method. Pasting through drag and drop is set up in the JQuery UI drop method configured in this file.